src/Symfony/Component/Yaml/Inline.php

```diff
@@ -471,64 +472,35 @@ private static function parseMapping(string $mapping, int $flags, int &$i = 0, a
-                        } elseif (isset($output[$key])) {
-                            throw new ParseException(sprintf('Duplicate key "%s" detected.', $key), self::$parsedLineNumber + 1, $mapping);
-                        }
                         break;
                     case '{':
                         // nested mapping
                         $value = self::parseMapping($mapping, $flags, $i, $references);
-                        // Spec: Keys MUST be unique; first one wins.
-                        // Parser cannot abort this mapping earlier, since lines
-                        // are processed sequentially.
-                        // But overwriting is allowed when a merge node is used in current block.
-                        if ('<<' === $key) {
-                            $output += $value;
-                        } elseif ($allowOverwrite || !isset($output[$key])) {
-                            if (null !== $tag) {
-                                $output[$key] = new TaggedValue($tag, $value);
-                            } else {
-                                $output[$key] = $value;
-                            }
-                        } elseif (isset($output[$key])) {
-                            throw new ParseException(sprintf('Duplicate key "%s" detected.', $key), self::$parsedLineNumber + 1, $mapping);
-                        }
                         break;
                     default:
                         $value = self::parseScalar($mapping, $flags, [',', '}', "\n"], $i, null === $tag, $references);
-                        // Spec: Keys MUST be unique; first one wins.
-                        // Parser cannot abort this mapping earlier, since lines
-                        // are processed sequentially.
-                        // But overwriting is allowed when a merge node is used in current block.
-                        if ('<<' === $key) {
-                            $output += $value;
-                        } elseif ($allowOverwrite || !isset($output[$key])) {
-                            if (null !== $tag) {
-                                $output[$key] = new TaggedValue($tag, $value);
-                            } else {
-                                $output[$key] = $value;
-                            }
-                        } elseif (isset($output[$key])) {
-                            throw new ParseException(sprintf('Duplicate key "%s" detected.', $key), self::$parsedLineNumber + 1, $mapping);
-                        }
                         --$i;
+                        break;
                 }
+
                 ++$i;
 
+                // Spec: Keys MUST be unique; first one wins.
+                // Parser cannot abort this mapping earlier, since lines
+                // are processed sequentially.
+                // But overwriting is allowed when a merge node is used in current block.
+                if ('<<' === $key) {
+                    $output += $value;
```

Comment:
the code changed in case of nested sequences now. Is that expected ?

Reply:
Reverted; initially I thought all three cases' value concatenation were duplicates, but only the latter two are. There's little gained by getting rid of this duplication since it would have to introduce a condition - so I abandoned this particular part of the change.